syft/source_providers.go

```diff
@@ -0,0 +1,52 @@
+package syft
+
+import (
+	"fmt"
+
+	"github.com/anchore/stereoscope/tagged"
+	"github.com/anchore/syft/syft/source"
+	"github.com/anchore/syft/syft/source/directory"
+	"github.com/anchore/syft/syft/source/filesource"
+	"github.com/anchore/syft/syft/source/stereoscope"
+)
+
+// SourceProviders returns all the configured source providers known to syft
+func SourceProviders(sourceProviderConfig ...SourceProviderConfig) tagged.Values[source.Provider] {
```

Comment:
something doesn't quite feel right with having a stereoscope primitive (`tagged*`) bring raised up on the top-level syft api